### PR TITLE
D8 nid 486

### PR DESCRIPTION
**Pull request by DuttonMa**
Add 'revision number' column to the revisions tab for all content types

### CODE REVIEW

File: origins_common/origins_common.module [hunk 1]
```diff
@@ -85,3 +85,59 @@ function origins_common_form_entity_embed_dialog_alter(&$form, FormStateInterfac
...
+  //
+  // Making this change in preprocess is not beautiful, but seems
+  // to be the only workable solution as the 'diff' module has
+  // already been enabled and has taken over the core route with
+  // its own controller. If we created our own controller to do
+  // this then we would have to call the contrib controller,
+  // which does not sound like a good idea.
+  if (\Drupal::routeMatch()->getRouteName() == 'entity.node.version_history') {
+    // Add 'Revision Number' column header.
+    $newhead = ['tag' => 'th', 'content' => ['#markup' => t('Revision Number')]];
```
Comment (omahm) [inline origins_common/origins_common.module]: Need to switch this to snake case.

File: origins_common/origins_common.module [hunk 2]
```diff
@@ -85,3 +85,59 @@ function origins_common_form_entity_embed_dialog_alter(&$form, FormStateInterfac
...
+    // (and this code should work whether the 'diff' module has been enabled or
+    // not).
+    array_splice($variables['header'], 1, 0, [$newhead]);
+    if (isset($variables['header']['select_column_one'])) {
+      $variables['header']['select_column_one']['content'] = ['#markup' => t('Compare 1')];
+    }
+    if (isset($variables['header']['select_column_two'])) {
+      $variables['header']['select_column_two']['content'] = ['#markup' => t('Compare 2')];
+    }
+    // Loop through the rows.
```
Comment (omahm) [inline origins_common/origins_common.module]: Irrelevant comment, we can see its looping though the rows. Pad this out to explain why you're looping the rows.

File: origins_common/origins_common.module [hunk 3]
```diff
@@ -85,3 +85,59 @@ function origins_common_form_entity_embed_dialog_alter(&$form, FormStateInterfac
...
+      else {
+        // There is no 'revert' button in this row,
+        // therefore it must be the current version.
+        //
+        // Cannot manage to extract the revision id from this row
+        // anywhere so we'll (reluctantly) have to load up the full node.
+        $node = \Drupal::routeMatch()->getParameter('node');
+        $this_version = $node->getRevisionId();
+      }
+      $newcol = ['tag' => 'td', 'content' => ['#markup' => $this_version]];
```
Comment (omahm) [inline origins_common/origins_common.module]: Snake case

File: origins_common/origins_common.module [hunk 4]
```diff
@@ -85,3 +85,59 @@ function origins_common_form_entity_embed_dialog_alter(&$form, FormStateInterfac
...
+      // Insert this new column after the first one.
+      // Use array_splice as we don't know exactly how many columns there
+      // will be (and this code should work whether the 'diff' module has
+      // been enabled or not).
+      array_splice($variables['rows'][$idx]['cells'], 1, 0, [$newcol]);
```
Comment (omahm) [inline origins_common/origins_common.module]: Snake case.

File: origins_common/origins_common.module (hunk 1 above)
Comment (johangant) [inline origins_common/origins_common.module]: Something to be mindful of with labels and such like is consistency around Sentence case and Title case. Essentially, either is fine but keeping things consistent makes a big difference to the UX.

See https://uxplanet.org/why-letter-casing-is-important-to-consider-during-design-decisions-50402acd0a4e for details.

Personally, I'm all for sentence case as I find it much, much easier to read... but that's just me. Apple sees things differently here... but who listens to them!? 🤣

File: origins_common/origins_common.module [hunk 5]
```diff
@@ -85,3 +85,59 @@ function origins_common_form_entity_embed_dialog_alter(&$form, FormStateInterfac
...
+  if (\Drupal::routeMatch()->getRouteName() == 'entity.node.version_history') {
+    // Add 'Revision Number' column header.
+    $newhead = ['tag' => 'th', 'content' => ['#markup' => t('Revision Number')]];
+    // Insert this new column after the first one.
+    // Use array_splice as we don't know exactly how many columns there will be
+    // (and this code should work whether the 'diff' module has been enabled or
+    // not).
+    array_splice($variables['header'], 1, 0, [$newhead]);
+    if (isset($variables['header']['select_column_one'])) {
+      $variables['header']['select_column_one']['content'] = ['#markup' => t('Compare 1')];
```
Comment (johangant) [inline origins_common/origins_common.module]: Haven't seen this in-situ yet, but is there a more descriptive label than 'Compare 1' and 'Compare 2'? What does the 1 and 2 refer to?

File: origins_common/origins_common.module (hunk 1 above)
Comment (DuttonMa) [reply origins_common/origins_common.module]: I agree with you, I prefer sentence case too generally, have corrected this as suggested

File: origins_common/origins_common.module (hunk 2 above)
Comment (DuttonMa) [reply origins_common/origins_common.module]: Fixed

File: origins_common/origins_common.module (hunk 4 above)
Comment (DuttonMa) [reply origins_common/origins_common.module]: Fixed

File: origins_common/origins_common.module (hunk 1 above)
Comment (DuttonMa) [reply origins_common/origins_common.module]: Fixed

File: origins_common/origins_common.module (hunk 5 above)
Comment (DuttonMa) [reply origins_common/origins_common.module]: Well, the 'diff' module adds a 'Compare selected revisions' button at the bottom of the form and then just adds two columns of radios. No headings are added to the columns but the user is obviously expected to click one in each column and then hit the compare button. I thought that something was needed (especially after I added a new column and made the page more busy) so I came up with 'Compare 1' and 'Compare 2' - I'm not delighted with them so am open to suggestions !

Comment (DuttonMa) [reply origins_common/origins_common.module]: Have now gone for 'Compare revision 1' and 'Compare revision 2' as agreed

**Pull request merged**